nifi-nar-bundles/nifi-standard-bundle/nifi-transform-json-ui/src/main/java/org/apache/nifi/web/standard/api/processor/ProcessorResource.java

```diff
@@ -0,0 +1,100 @@
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.apache.nifi.web.ComponentDetails;
+import org.apache.nifi.web.ComponentDescriptor;
+import org.apache.nifi.web.NiFiWebConfigurationContext;
+
+import org.apache.nifi.web.NiFiWebConfigurationRequestContext;
+import org.apache.nifi.web.standard.api.AbstractStandardResource;
+import org.apache.nifi.web.util.CustomUIUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Path("/standard/processor")
+public class ProcessorResource extends AbstractStandardResource {
+
+    private static final Logger logger = LoggerFactory.getLogger(ProcessorResource.class);
+
+    @GET
+    @Produces({MediaType.APPLICATION_JSON})
+    @Path("/properties")
+    public Response getProperties(@QueryParam("processorId") final String processorId) {
+        final NiFiWebConfigurationContext nifiWebContext = getWebConfigurationContext();
+        final ComponentDetails componentDetails = CustomUIUtils.getComponentDetails(nifiWebContext, processorId, request);
+        final Response.ResponseBuilder response = CustomUIUtils.applyCacheControl(Response.ok(componentDetails.getProperties()));
+        return response.build();
+    }
+
+    @GET
+    @Produces({MediaType.APPLICATION_JSON})
+    @Path("/type")
+    public Response getType(@QueryParam("processorId") final String processorId) {
+        final NiFiWebConfigurationContext nifiWebContext = getWebConfigurationContext();
+        final ComponentDetails componentDetails = CustomUIUtils.getComponentDetails(nifiWebContext, processorId, request);
+        final Response.ResponseBuilder response = CustomUIUtils.applyCacheControl(Response.ok(componentDetails.getType()));
+        return response.build();
+    }
+
+    @GET
+    @Produces({MediaType.APPLICATION_JSON})
+    @Path("/descriptors")
+    public Response getDescriptors(@QueryParam("processorId") final String processorId) {
+        final NiFiWebConfigurationContext nifiWebContext = getWebConfigurationContext();
+        final ComponentDetails componentDetails = CustomUIUtils.getComponentDetails(nifiWebContext, processorId, request);
+        final Map<String,ComponentDescriptor> descriptors = componentDetails.getDescriptors();
+        final Response.ResponseBuilder response = CustomUIUtils.applyCacheControl(Response.ok(descriptors));
+        return response.build();
+    }
+
+    @GET
+    @Produces({MediaType.APPLICATION_JSON})
+    @Path("/details")
+    public Response getDetails(@QueryParam("processorId") final String processorId) {
+        final NiFiWebConfigurationContext nifiWebContext = getWebConfigurationContext();
+        final ComponentDetails componentDetails = CustomUIUtils.getComponentDetails(nifiWebContext, processorId, request);
+        final Response.ResponseBuilder response = CustomUIUtils.applyCacheControl(Response.ok(componentDetails));
+        return response.build();
+    }
+
+    @POST
```

Comment:
Is this endpoint creating a single new property? Or is it updating the specified processor's properties with those listing? Assuming the latter, this should probably be a PUT.